src/datasets/arrow_dataset.py

```diff
@@ -1859,9 +1859,14 @@ def map(
                 `batch_size <= 0` or `batch_size == None`: Provide the full dataset as a single batch to `function`.
             drop_last_batch (:obj:`bool`, default `False`): Whether a last batch smaller than the batch_size should be
                 dropped instead of being processed by the function.
-            remove_columns (`Optional[Union[str, List[str]]]`, default `None`): Remove a selection of columns while doing the mapping.
+            remove_columns (`Optional[Union[bool,List[str]]]`, defaults to `None`): Remove a selection of columns while doing the mapping.
```

Comment:
So I thought about this, and this would mean a breaking change. Right now, `False` and `None` do the same thing, but some people might be setting that value to `None`.